tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -142,7 +149,147 @@ def test_brand_string(test_microvm_with_api, network_config):
 @pytest.mark.skipif(
     PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
 )
-@pytest.mark.parametrize("cpu_template", ["T2", "C3"])
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
+    reason="CPU templates are only available on Intel.",
+)
+@pytest.mark.skipif(
+    utils.get_kernel_version(level=1) not in SUPPORTED_KERNELS,
+    reason=f"Supported kernels are {SUPPORTED_KERNELS}",
+)
+@pytest.mark.parametrize("cpu_template", ["T2S"])
+@pytest.mark.timeout(900)
+@pytest.mark.nonci
+def test_cpu_rdmsr(bin_cloner_path, network_config, cpu_template):
+    """
+    Test MSRs that are available to the Guest.
+
+    This test boots a Firecracker uVM and tries to read a set of MSRs from the guest.
+    The Guest MSR list is compared against a list of MSRs that are expected when running
+    on a particular host kernel and with a particular Guest CPU template.
+    The list is different for each kernel version because Firecracker relies on
+    MSR emulation provided by KVM. If KVM emulation changes, then the MSR list
+    available to the guest might change also.
+    The list is also dependant on CPUID (CPU templates) since some MSRs are not available
+    if CPUID features are disabled.
+    Lastly, this tests also checks for MSR values against the baseline. This helps validate
+    that defaults have not changed due to emulation implementation changes in the kernel.
+
+    TODO: This only validates T2S templates. Since T2 and C3 did not set the
+    ARCH_CAPABILITIES MSR, the value of that MSR is different between different
+    host CPU types (see Github PR #3066). So we can either:
+        * add an exceptions for different template types when checking values
+        * deprecate T2 and C3 since they are somewhat broken
+
+    @type: functional
+    """
+
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Testing matrix:
+    # - Guest kernel: Linux 4.14 & Linux 5.10
+    # - Rootfs: Ubuntu 18.04 with msr-tools package installed
+    # - Microvm: 1vCPU with 1024 MB RAM
+    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="1vcpu_1024mb"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels())
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="bionic-msrtools"))
+    assert len(disk_artifacts) == 1
+
+    test_context = TestContext()
+    test_context.custom = {
+        "builder": MicrovmBuilder(bin_cloner_path),
+        "network_config": network_config,
+        "cpu_template": cpu_template,
+    }
+    test_matrix = TestMatrix(
+        context=test_context,
+        artifact_sets=[microvm_artifacts, kernel_artifacts, disk_artifacts],
+    )
+    test_matrix.run_test(_test_cpu_rdmsr)
+
+
+def _test_cpu_rdmsr(context):
```

Comment:
`_` is used as a prefix in Rust and often elsewhere to mean an item is unused, possibly giving the impression that `_test_cpu_rdmsr` is unused.

I would suggest a name change here, possibly  `test_cpu_rdmsr_internal` or `test_cpu_rdmsr_fn`.

Reply:
`_` is a standard Python prefix for things that are not part of the public API.

Reply:
I forgot this was a thing, you're right.